lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -310,8 +262,12 @@
 
     platform.registerFunction(platform.FunctionKind.Array_sort, function (compareFn) {
         //#sec-array.prototype.sort
-        if (compareFn !== undefined && typeof compareFn !== "function") {
-            __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.sort");
+        if (compareFn !== undefined) {
+            if (typeof compareFn !== "function") {
+                __chakraLibrary.raiseFunctionArgument_NeedFunction("Array.prototype.sort");
+            }
+        } else {
+            compareFn = __chakraLibrary.DefaultSortCompare;
```

Comment:
Moved the default compare function assignment up here to avoid an extra condition below.